azure-pipelines.yml

```diff
@@ -213,3 +213,38 @@ jobs:
       continueOnError: true
       condition: not(succeeded())
 
+- job: Correctness_Rebuild
+  pool:
+    vmImage: windows-2019
+  timeoutInMinutes: 90
+  steps:
+    - template: eng/pipelines/checkout-windows-task.yml
+
+    - task: PowerShell@2
+      displayName: Restore
+      inputs:
+        filePath: eng/build.ps1
+        arguments: -configuration Debug -prepareMachine -ci -restore -binaryLog
+
+    - task: PowerShell@2 
+      displayName: Build
+      inputs:
+        filePath: eng/build.ps1
+        arguments: -configuration Debug -prepareMachine -ci -build -bootstrap -publish -binaryLog -skipDocumentation
+
+    - script: .\artifacts\bin\BuildValidator\Debug\net472\BuildValidator.exe --assembliesPath .\artifacts\obj\Microsoft.CodeAnalysis --debugPath .\artifacts\BuildValidator --sourcePath .
+      displayName: Run BuildValidator
```

Comment:
It'd be better to have this logic in a `eng/validate-rebuild.ps1` file, so it can be easily run locally.

Reply:
Actually, why not call test-rebuild.ps1?

---
In reply to: [578014609](https://github.com/dotnet/roslyn/pull/50968#discussion_r578014609) [](ancestors = 578014609)